lib/datetime/datetime.ex

```diff
@@ -789,7 +789,7 @@ defmodule Timex.DateTime do
       value == -12 -> %{datetime | :year => year - 1}
       m == 0 -> %{datetime | :year => year - 1, :month => 12}
       m > 12 -> %{datetime | :year => year + div(m, 12), :month => rem(m, 12)}
-      m < 0  -> %{datetime | :year => year + div(m, 12), :month => 12 + rem(m, 12)}
+      m < 0  -> %{datetime | :year => year + Enum.min([div(m, 12), -1]), :month => 12 + rem(m, 12)}
```

Comment:
You can change `Enum.min([div(m, 12), -1])` to `min(div(m, 12), -1)`, the latter should be much faster since it's a simple comparison of two integer values, rather than constructing a list, etc.